### PR TITLE
Feature/258 sunSearch

### LINKED ISSUE #258: sunSearch module
**Describe your use case**
This guidance module performs outputs the reference rates and accelerations required to perform three rotations about three principal body axes, sweeping user-specified angles. The output rate and acceleration profiles for each independent rotation correspond to those of a bang-bang type of maneuver, with a coasting arc in the middle. The module requires the user to define the maximum deliverable torques about each axis, each rotation angle, the maximum slew rate allowed for each rotation, and the time to perform each rotation. Because the problem is inherently overly prescribed, not all the requirements can be satisfied simultaneously. 

**Describe alternatives solutions you've considered**
n/a

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by rcalaon**
* **Tickets addressed:** bsk-258
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description
This module computes three consecutive bang-bang rate and acceleration profiles for 3 rotations about any combination of principal body axes. Repetitions of the same axis are allowed. The output is a guidance message in which the reference rate is subtracted from the spacecraft body rate to close the control loop. Depending on the CSS configuration, different rotation sequences sweeping different angles might be required in order to ensure that the Sun is found.

## Verification
A UnitTest is provided to check the accuracy of the output.

## Documentation
New documentation is provided with this module.

## Future work
The module can be enhanced with more than 3 rotations if needed.

### CODE REVIEW

File: src/fswAlgorithms/attGuidance/sunSearch/sunSearch.cpp
Comment (patkenneally) [inline src/fswAlgorithms/attGuidance/sunSearch/sunSearch.cpp]: :large_orange_diamond: Lower case function name.

File: src/fswAlgorithms/attGuidance/sunSearch/sunSearch.cpp
Comment (patkenneally) [inline src/fswAlgorithms/attGuidance/sunSearch/sunSearch.cpp]: :large_orange_diamond: this->compute...

File: src/fswAlgorithms/attGuidance/sunSearch/sunSearch.cpp
Comment (patkenneally) [inline src/fswAlgorithms/attGuidance/sunSearch/sunSearch.cpp]: :large_orange_diamond: this->computeRef...

File: src/fswAlgorithms/attGuidance/sunSearch/sunSearch.cpp
Comment (patkenneally) [inline src/fswAlgorithms/attGuidance/sunSearch/sunSearch.cpp]: :large_orange_diamond: lower case function name

File: src/fswAlgorithms/attGuidance/sunSearch/sunSearch.cpp
Comment (patkenneally) [inline src/fswAlgorithms/attGuidance/sunSearch/sunSearch.cpp]: :large_orange_diamond: These time variables can all be lower case `localSimSeconds`, `currentSimNanos`

File: src/fswAlgorithms/attGuidance/sunSearch/sunSearch.h
Comment (patkenneally) [inline src/fswAlgorithms/attGuidance/sunSearch/sunSearch.h]: :large_orange_diamond: `CurrentSimNanos` can be `currentSimNanos`

File: src/fswAlgorithms/attGuidance/sunSearch/sunSearch.h
Comment (patkenneally) [inline src/fswAlgorithms/attGuidance/sunSearch/sunSearch.h]: :large_orange_diamond: What are your thoughts on refactoring each of these variables into a struct. That way the various methods that use them will not have to access each value with `index` everywhere. Happy to chat.

**Review by patkenneally — APPROVED** [review]
Legend:
:small_red_triangle_down: *Issues* to address before merge
:large_orange_diamond: *Requests* that should not block merge, but should at least be discussed
:large_blue_circle: *Recommendations* that can be ignored if desired

Nice! Huge work. Let's chat about my little struct thought.

File: src/fswAlgorithms/attGuidance/sunSearch/sunSearch.cpp
Comment (rcalaon) [reply src/fswAlgorithms/attGuidance/sunSearch/sunSearch.cpp]: Done

File: src/fswAlgorithms/attGuidance/sunSearch/sunSearch.cpp
Comment (rcalaon) [reply src/fswAlgorithms/attGuidance/sunSearch/sunSearch.cpp]: Done

File: src/fswAlgorithms/attGuidance/sunSearch/sunSearch.cpp
Comment (rcalaon) [reply src/fswAlgorithms/attGuidance/sunSearch/sunSearch.cpp]: Done

File: src/fswAlgorithms/attGuidance/sunSearch/sunSearch.cpp
Comment (rcalaon) [reply src/fswAlgorithms/attGuidance/sunSearch/sunSearch.cpp]: Done

File: src/fswAlgorithms/attGuidance/sunSearch/sunSearch.cpp
Comment (rcalaon) [reply src/fswAlgorithms/attGuidance/sunSearch/sunSearch.cpp]: Done

File: src/fswAlgorithms/attGuidance/sunSearch/sunSearch.h
Comment (rcalaon) [reply src/fswAlgorithms/attGuidance/sunSearch/sunSearch.h]: Done

File: src/fswAlgorithms/attGuidance/sunSearch/sunSearch.h
Comment (rcalaon) [reply src/fswAlgorithms/attGuidance/sunSearch/sunSearch.h]: I've combined all the properties into one big structure. Much cleaner overall.
I left out the inertias and the maximum torques because they pertain to each principal body axis, and not necessarily to the requested slews.

Comment (patkenneally) [reply src/fswAlgorithms/attGuidance/sunSearch/sunSearch.h]: Solid.

**Pull request merged**
